src/EnergyPlus/DuctLoss.cc

```diff
@@ -0,0 +1,1060 @@
+    {
+        Real64 MassFlowRate;
+        Real64 Tamb;
+        Real64 Wamb;
+        Real64 Tin;
+        Real64 Tout;
+        Real64 Win;
+        Real64 Wout;
+        Real64 UA;
+        Real64 CpAir;
+        Real64 enthalpy;
+        int NodeNum1;
+        int NodeNum2;
+        int NodeNum;
+        int TypeNum = state.afn->AirflowNetworkCompData(state.afn->AirflowNetworkLinkageData(this->LinkageNum).CompNum).TypeNum;
+        Real64 DuctSurfArea = state.afn->DisSysCompDuctData(TypeNum).L * state.afn->DisSysCompDuctData(TypeNum).hydraulicDiameter * Constant::Pi;
+        Real64 UThermal = state.afn->DisSysCompDuctData(TypeNum).UThermConduct;
+        Real64 UMoisture = state.afn->DisSysCompDuctData(TypeNum).UMoisture;
+
+        this->Qsen = 0.0;
+        this->Qlat = 0.0;
+
+        if (this->EnvType == EnvironmentType::Zone && this->ZoneNum > 0) {
+            Tamb = state.dataZoneTempPredictorCorrector->zoneHeatBalance(this->ZoneNum).MAT;
+            Wamb = state.dataZoneTempPredictorCorrector->zoneHeatBalance(this->ZoneNum).airHumRat;
+        } else {
+            Tamb = this->tambSched->getCurrentVal();
+            Wamb = this->wambSched->getCurrentVal();
+        }
+        NodeNum1 = state.afn->AirflowNetworkLinkageData(this->LinkageNum).NodeNums[0];
+        NodeNum2 = state.afn->AirflowNetworkLinkageData(this->LinkageNum).NodeNums[1];
+
+        switch (this->LossSubType) {
+        case DuctLossSubType::SupplyTrunk: {
+            NodeNum = state.afn->DisSysNodeData(NodeNum1).EPlusNodeNum;
+            MassFlowRate = state.dataLoopNodes->Node(NodeNum).MassFlowRate;
+            Tin = state.dataLoopNodes->Node(NodeNum).Temp;
+            Win = state.dataLoopNodes->Node(NodeNum).HumRat;
+            state.afn->AirflowNetworkNodeSimu(NodeNum1).TZ = Tin;
+            state.afn->AirflowNetworkNodeSimu(NodeNum1).WZ = Win;
+            CpAir = Psychrometrics::PsyCpAirFnW(state.dataLoopNodes->Node(NodeNum).HumRat);
+            Tout = Tamb + (Tin - Tamb) * General::epexp(-UThermal * DuctSurfArea, (MassFlowRate * CpAir));
+            Wout = Wamb + (Win - Wamb) * General::epexp(-UMoisture * DuctSurfArea, MassFlowRate);
+            this->Qsen = -MassFlowRate * CpAir * (Tamb - Tin) * (1.0 - General::epexp(-UThermal * DuctSurfArea, (MassFlowRate * CpAir)));
+            this->Qlat = -MassFlowRate * (Wamb - Win) * (1.0 - General::epexp(-UMoisture * DuctSurfArea, MassFlowRate));
+            state.afn->AirflowNetworkNodeSimu(NodeNum2).TZ = Tout;
+            state.afn->AirflowNetworkNodeSimu(NodeNum2).WZ = Wout;
+            if (!state.dataDuctLoss->SubTypeSimuFlag(int(DuctLossSubType::SupplyBranch) + 1)) {
+                enthalpy = Psychrometrics::PsyHFnTdbW(Tout, Wout);
+                for (int OutNodeNum = 1; OutNodeNum <= state.dataSplitterComponent->SplitterCond(1).NumOutletNodes; OutNodeNum++) {
+                    state.dataLoopNodes->Node(state.dataSplitterComponent->SplitterCond(1).OutletNode(OutNodeNum)).Temp = Tout;
```

Comment:
Why a hard subscript of 1 here? Same ? below for `MixerCond(1)`.

Reply:
> The present deliverable covers a single AirLoop and allows a single Splitter and Mixer. I am going to make more general for future more coverage.

It would be better to have a variable for splitterNum and mixerNumand initialize them to 1  (or get this from the system num?).